10_08_2017/perturbed_big_disp.py

```diff
@@ -0,0 +1,453 @@
+        elif min(d, d*b)< np.real(W**2) <max(d, d*b):
+            return disp_rel_1_2_a0b0(W)
+        elif  max(d, d*b) < np.real(W**2):
+            return np.NAN #disp_rel_1_4(ga0, gb0, gae, gbe, W)
+        else:
+            return np.NaN
+    elif  np.real(W**2) > max(1,b) and np.real(W**2)> max(d, d*b):
+        return disp_rel_1_2_aebe(W)
+    else:
+        return np.NaN
+        
+def big_dr_0_deriv(W):
+    if np.real(W**2) < b / (b+1):
+        if np.real(W**2) < d * b / (b+1):
+            return deriv_dr_0_2_aeb0(W)
+        elif  d * b / (b+1) < np.real(W**2) < min(d, d*b): 
+            return deriv_dr_0_2_beb0(W)
+        elif  max(d, d*b) < np.real(W**2):
+            return np.NaN #disp_rel_1_3(gbe, gae, gb0, W)
+        else:
+            return np.NaN
+    elif  b / (b+1) < np.real(W**2) < min(1, b):
+        if np.real(W**2) < d * b / (b+1):
+            return deriv_dr_0_2_aea0(W)
+        elif  d * b / (b+1) < np.real(W**2) < min(d, d*b): 
+            return deriv_dr_0_2_bea0(W)
+        elif  max(d, d*b) < np.real(W**2):
+            return np.NAN #disp_rel_1_3(gbe, gae, ga0, W)
+        else:
+            return np.NaN
+    elif min(1, b) < np.real(W**2) < max(1,b):
+        if np.real(W**2) < d * b / (b+1):
+            return np.NAN #disp_rel_1_3(ga0, gb0, gae, W)
+        elif  d * b / (b+1) < np.real(W**2) < min(d, d*b): 
+            return np.NAN #disp_rel_1_3(ga0, gb0, gbe, W)
+        elif min(d, d*b)< np.real(W**2) <max(d, d*b):
+            return deriv_dr_0_2_a0b0(W)
+        elif  max(d, d*b) < np.real(W**2):
+            return np.NAN #disp_rel_1_4(ga0, gb0, gae, gbe, W)
+        else:
+            return np.NaN
+    elif  np.real(W**2) > max(1,b) and np.real(W**2) > max(d, d*b):
+        return deriv_dr_0_2_aebe(W)
+    else:
+        return np.NaN
+
+def perturbed(W):
+    W1 = -big_disp_1(W) / big_dr_0_deriv(W)
+    return W + theta * W1
+
+b_num=100
+b_start = -2
+b_end = 2
+b_range=np.logspace(b_start,b_end,b_num)
+b=np.logspace(b_start,b_end,b_num)
+#d=np.linspace(b_start, b_end,b_num)
+
+snum=200
+s_start =0
+s_end=4
+s_range = np.linspace(s_start,s_end,snum)
+
+n_0roots=50
+megaroots=np.nan*np.ones(shape=(n_0roots,b_num),dtype=np.complex64)
+
+
+#Check it's not finding the same root multiple times
+def check(root_temp, root):
+    n = True
+    if len(root_temp)==0:
+        root_temp.append(root)
+    else:
+        for x in root_temp:
+            if np.abs(np.real(x) - np.real(root)) < 10**(-6) and \
+               np.abs(np.imag(x) - np.imag(root)) < 10**(-6):
+                   n = False
+            else:
+                pass
+        if n == True:
+            root_temp.append(root)
+
+# find list of roots
+root_list = []
+for b in b_range:
+    root_temp = []
+    for s in s_range:
+        try:
+            root = sp.newton(big_disp_0, s + 0.01*1j, tol = 10**(-10))
+            print root
+            check(root_temp, root)
+        except Exception as error:
```

Comment:
You should be careful with this because `except` catches subclasses of errors, and all errors are derived from the `Exception` class. So this will still catch all errors, just like `except:` on it's own would. Printing the error at least means it's not silent, but you should probably still use a more specific error (you can [define your own](https://docs.python.org/2/tutorial/errors.html#tut-userexceptions) if there isn't an appropriate Python one).